modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryAdapter.java

```diff
@@ -654,12 +656,30 @@ private boolean isSafeLossPolicy() {
                 if (prj != null || part != null)
                     return nodes(cctx, prj, part);
```

Comment:
Does the case with explicit partition work correctly after made changes? Previously we had a single node _projection_ for REPLICATED caches, but now not.

Reply:
I've added check for query by partition in the test. It works.